### DISCUSSION TITLE
Minimal LogicAnalyzer

### DISCUSSION THREAD

**Discussion by bablokb** [gusmanb/logicanalyzer#69] [Show and tell]
Hi,

this is FYI only: I created a 3D-printed case for a minimal LogicAnalyzer: https://github.com/bablokb/3D-pico-lipo-shim

Only 8 channels and no level-shifting, but sufficient for my needs. It features a Pico-W and a LiPo attached using a LiPo-Shim from Pimoroni.

**Comment by gusmanb** [comment]
(I have moved the post to discussions so others will be able to find it 😄)

Pretty cool!

Btw, have you tested how much time the W lasts with the battery? I would be interested on knowing it.

**Reply by bablokb** [reply]
Not yet. But I did a number of measurements in a different context, see https://github.com/bablokb/pcb-pico-datalogger/blob/main/power.md

**Comment by bablokb** [comment]
Here is a measurement I did with the Pico-W (connect, configure, 2x I2C capture + analysis). Sampling is with 100/s, so the graph does not catch every short spike as you  in fact have with WLAN. Some current spikes are hidden below the respective voltage drops.

Net result: I have about 60mA mean current, so my little 350mAh LiPo should last five hours.

![LogicAnalyzer](https://github.com/gusmanb/logicanalyzer/assets/9638355/24ee80ff-03c9-4eae-b826-86d625df8e97)

**Reply by gusmanb** [reply]
Excellent! 

Thanks a lot for the info, I will add this to the wiki when I update it for the next release ( V5.0 is comming very soon with really neat features 😄 ) as it's really useful. 

One of the main goals of making the analyzer compatible with the W was precisely to be able to have a completely standalone analyzer with no wires at all except for the ones for the channel inputs, so it should be powered with a battery like you did and with this now I know that with a very cheap battery the device can last up to 5 hours, more than enough. 

I need to start thinking in creating a PCB that includes the battery management, maybe for release 6 I could add it.

Cheers!

**Comment by bablokb** [comment]
I started with the normal pico, but it adds the need for another cable so the W is a much better choice. I can keep it near the circuit under test.

In fact it would be great if the GUI would tell me about  the battery level. The pico  can measure VSYS voltage, using GPIO29/ADC3 giving VSYS/3. Not very exact, but good enough  to know when to recharge.

**Reply by gusmanb** [reply]
Do you have any working example for the W when the wifi is connected? I tried it on another project and it completelly messed up the wifi comms as GPIO25 and GPIO29 are used for the wifi module, I tried to change the mode and revert it back to its original state but some times the wifi hung up...

**Comment by bablokb** [comment]
For most of my project I am using CircuitPython. And they have the following code: https://github.com/adafruit/circuitpython/blob/main/ports/raspberrypi/common-hal/analogio/AnalogIn.c#L82. Of course that is only the high-level part with the switching of the state, but maybe this helps. Also have a look at line 39ff

**Reply by gusmanb** [reply]
Hmmm, they are disabling the interrupts to do the measurements, that may be the problem I had, possibly the module was triggering an int, the driver got the control and as the pins were measuring the voltage that messed up the comms... Ok, I will try it and if it works I will add the capability :)

**Comment by gusmanb** [comment]
Hi.

I have prepared a firmware and app with the power info. It will read the voltage and VBUS to show if the device is powered by battery or it is being charged. It will poll the info each 30 seconds when it is in idle, when a capture is running it will stop polling and will retrieve the info immediatelly after the capture. The battery info will be shown only if the device is connected through WiFi, it does not make sense to show it if you have connected it through USB ;)
The info is shown in the info pane, at the bottom right.

As far as I have tested all has been working correctly, it seems that the trick was the interrupt disabling :)

You can download it from [here](https://mega.nz/file/3WAjmLBL#TprSgGWDlg789Ibkx13d8dV5TEGAzeThJXWmWTsWM04)

Cheers!

**Reply by bablokb** [reply]
Thanks! I won't be able to test before Thursday but I will certainly do as soon as I am back

**Comment by bablokb** [comment]
Works perfectly! A very useful feature, thank you for implementing this.

**Reply by gusmanb** [reply]
Excellent, I will include it in the next release :)
